src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -223,17 +227,16 @@ private async Task CloseMessageReceiverAsync()
 
             try
             {
-                Logger.LogTrace("Closing message pump '{JobId}' on entity path '{EntityPath}' in '{Namespace}'",  JobId, EntityPath, Namespace);
-                await _messageProcessor.StopProcessingAsync();
+                Logger.LogTrace("Closing message pump '{JobId}' on entity path '{EntityPath}' in '{Namespace}'", JobId, EntityPath, Namespace);
+                await _messageProcessor.CloseAsync();
                 _messageProcessor.ProcessMessageAsync -= ProcessMessageAsync;
                 _messageProcessor.ProcessErrorAsync -= ProcessErrorAsync;
-                await _messageProcessor.CloseAsync();
-                Logger.LogInformation("Message pump '{JobId}' on entity path '{EntityPath}' in '{Namespace}' closed : {Time}",  JobId, EntityPath, Namespace, DateTimeOffset.UtcNow);
+               
+                Logger.LogInformation("Message pump '{JobId}' on entity path '{EntityPath}' in '{Namespace}' closed : {Time}", JobId, EntityPath, Namespace, DateTimeOffset.UtcNow);
             }
-            catch (TaskCanceledException) 
+            catch (Exception exception) when (exception is TaskCanceledException || exception is OperationCanceledException)
             {
-                // Ignore.
-            } 
+            }
```

Comment:
I think a small comment to indicate that we're deliberately swallowing this exception because we're trying to stop gracefully would be appropriate :)
Or maybe a LogTrace or LogInformation call which describes this.

Reply:
Yes, that would be good 👍 .
It also seems that if we let this throw, the service still shuts down gracefully (due to the type of exception), it's only bc we also want to catch other types of exceptions that we have this.
Hmm, maybe we should add an exception filter to only catch in case it isn't one of those, that would be better, maybe.

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/313/commits/39abf09b93a637870cc5542689716bfc1242aa54